### PR TITLE
ENH: fast Bash tab completion

### LINKED ISSUE #6: fix bash tab completion performance issues
It's *really* slow right now.

### PR DESCRIPTION
**Pull request by jairideout**
See README.md or run the `qiime` command for instructions on how to enable Bash
tab completion.

This is basic tab completion supporting arbitrarily nested subcommands and options. More advanced completion (e.g. completion on .qza files when completing an action's inputs) are planned in the future. This works in a very general way by traversing the command tree structure starting at `q2cli.commands.RootCommand` and building a bash completion script. No specific knowledge of plugins, actions, or builtin commands is required, so updates to CLI commands should not require updates to the Bash completion code. The CLI cache manages this file so it will always match the latest deployment state. All users need to do is add a simple `eval` line to their `.bashrc`/`.bash_profile` and they are GTG!

Fixes #6.

### CODE REVIEW

File: q2cli/commands.py [hunk 1]
```diff
@@ -23,13 +23,29 @@ class RootCommand(click.MultiCommand):
         ('dev', q2cli.dev.dev)
     ])
 
+    def __init__(self, *args, **kwargs):
+        # 'help' is `None`, drop it to supply our own.
+        if 'help' in kwargs:
+            del kwargs['help']
+        super().__init__(help=ROOT_COMMAND_HELP, *args, **kwargs)
```
Comment (ebolyen) [inline q2cli/commands.py]: Why not do this on the root command annotation? `__main__:qiime`?

Comment (ebolyen) [inline q2cli/commands.py]: So any extra arguments passed to 
```python
@click.command(cls=q2cli.commands.RootCommand, invoke_without_command=True,
               no_args_is_help=True)
```
are sent to the constructor, which means you should be able to just add `help=` right there.

Comment (jairideout) [inline q2cli/commands.py]: Thanks I'll do that!

Comment (jairideout) [inline q2cli/commands.py]: I'll fix this in a future PR, we wanted to get this in so ppl can start playing around with it

**Comment by jairideout** [comment]
🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝 🚀 🎸 🌔 🌝

**Pull request merged**